openml_OS/models/api/v1/Api_data.php

```diff
@@ -294,11 +294,27 @@ private function data_edit() {
       $this->returnError( 1064, $this->version );
       return;
     }
-    // check if user owns dataset
+
+    // create a copy of the dataset with same file but different ID and different user ID
     if($dataset->uploader != $this->user_id and !$this->user_has_admin_rights) {
-      $this->returnError( 1065, $this->version );
+      $dataset->uploader = $this->user_id;
+      unset($dataset->did); 
+      $data_id = $this->Dataset->insert($dataset);
+
+      if (!$data_id) {
+      $this->returnError(1065, $this->version);
+      return;
+    }
+    }
+    // If sensitive fields need to be edited, create a copy with different ID
```

Comment:
And what about the second case? editing sensitive features like default target attribute? Do you think it is ok to create a copy in this case or should we allow the owner to edit it without creating these copies? As in edit them directly